hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/hdfs/server/federation/router/TestRouterRpc.java

```diff
@@ -1777,6 +1777,43 @@ public void testgetGroupsForUser() throws IOException {
     assertArrayEquals(group, result);
   }
 
+  @Test
+  public void testGetCachedDatanodeReport() throws Exception {
+    final DatanodeInfo[] datanodeReport =
+        routerProtocol.getDatanodeReport(DatanodeReportType.ALL);
+
+    // We should have 12 nodes in total
+    assertEquals(12, datanodeReport.length);
+
+    // We should be caching this information
+    DatanodeInfo[] datanodeReport1 =
+        routerProtocol.getDatanodeReport(DatanodeReportType.ALL);
+    assertArrayEquals(datanodeReport1, datanodeReport);
+
+    // Add one datanode
+    getCluster().getCluster().startDataNodes(getCluster().getCluster().getConfiguration(0),
```

Comment:
I think we need to clear state after the test case so that it won't affect others like `testNamenodeMetrics`.

Also long line.

Reply:
Done. Thanks. @sunchao